lib/Parse/Lexer.cpp

```diff
@@ -2298,46 +2298,34 @@ void Lexer::lexTrivia(syntax::TriviaList &Pieces, bool IsForTrailingTrivia) {
   // TODO: Handle invalid UTF8 sequence which is skipped in lexImpl().
   switch (*CurPtr++) {
   case '\n':
+    if (IsForTrailingTrivia)
+      break;
+    NextToken.setAtStartOfLine(true);
+    Pieces.appendOrSquash(TriviaPiece::newlines(1));
+    break;
```

Comment:
`goto Restart;`?

Reply:
I mistook.
I will fix it and add regression test.

Reply:
Ah I figured out why this passed existing tests.
After break, lexTrivia return with CurPtr reset.
Then in lexImpl, '\n' at CurPtr[0] is skipped as whitespace.
And when reenter lexTrivia, all state is valid for continue. 

I will fix it.